rust/operator-binary/src/airflow_db_controller.rs

```diff
@@ -236,15 +317,84 @@ fn build_init_job(
         env_var_from_secret("ADMIN_EMAIL", secret, "adminUser.email"),
         env_var_from_secret("ADMIN_PASSWORD", secret, "adminUser.password"),
+        EnvVar {
+            name: "PYTHONPATH".into(),
+            value: Some(LOG_CONFIG_DIR.into()),
+            ..Default::default()
+        },
+        EnvVar {
+            name: "AIRFLOW__LOGGING__LOGGING_CONFIG_CLASS".into(),
+            value: Some("log_config.LOGGING_CONFIG".into()),
+            ..Default::default()
+        },
     ];
 
-    let container = ContainerBuilder::new("airflow-init-db")
-        .expect("ContainerBuilder not created")
-        .image_from_product_image(resolved_product_image)
+    let mut containers = Vec::new();
+
+    let mut cb = ContainerBuilder::new("airflow-init-db").expect("ContainerBuilder not created");
+
+    cb.image_from_product_image(resolved_product_image)
         .command(vec!["/bin/bash".to_string()])
         .args(vec![String::from("-c"), commands.join("; ")])
         .add_env_vars(env)
-        .build();
+        .add_volume_mount("log-config", LOG_CONFIG_DIR)
+        .add_volume_mount("log", STACKABLE_LOG_DIR);
+
+    let mut volumes = Vec::new();
+
+    volumes.push(Volume {
+        name: "config".to_string(),
+        config_map: Some(ConfigMapVolumeSource {
+            name: Some(config_map_name.into()),
+            ..ConfigMapVolumeSource::default()
+        }),
+        ..Volume::default()
+    });
+    volumes.push(Volume {
+        name: "log".to_string(),
+        empty_dir: Some(EmptyDirVolumeSource {
+            medium: None,
+            size_limit: Some(Quantity(format!("{LOG_VOLUME_SIZE_IN_MIB}Mi"))),
+        }),
+        ..Volume::default()
+    });
+
+    if let Some(ContainerLogConfig {
+        choice:
+            Some(ContainerLogConfigChoice::Custom(CustomContainerLogConfig {
+                custom: ConfigMapLogConfig { config_map },
+            })),
+    }) = config.logging.containers.get(&Container::AirflowInitDb)
+    {
+        volumes.push(Volume {
+            name: "log-config".to_string(),
+            config_map: Some(ConfigMapVolumeSource {
+                name: Some(config_map.into()),
+                ..ConfigMapVolumeSource::default()
+            }),
+            ..Volume::default()
+        });
+    } else {
+        volumes.push(Volume {
+            name: "log-config".to_string(),
+            config_map: Some(ConfigMapVolumeSource {
+                name: Some(config_map_name.into()),
+                ..ConfigMapVolumeSource::default()
+            }),
+            ..Volume::default()
+        });
+    }
+
+    containers.push(cb.build());
+
+    if config.logging.enable_vector_agent {
+        containers.push(product_logging::framework::vector_container(
+            resolved_product_image,
+            "config",
+            "log",
+            config.logging.containers.get(&Container::Vector),
+        ));
+    }
```

Comment:
Common code moved to controller_commons.rs in 39fca612806acb8d34fdff046620b5e2e451ec4f

The following code was not consolidated because the config, container, and product image are different and the resulting code would not be significantly shorter:

```
    if config.logging.enable_vector_agent {
        containers.push(product_logging::framework::vector_container(
            resolved_product_image,
            "config",
            "log",
            config.logging.containers.get(&Container::Vector),
        ));
    }
```